runtime/tests/checker/external_mutation_test.go

```diff
@@ -0,0 +1,787 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package checker
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+func TestArrayUpdateIndexAccess(t *testing.T) {
+
+	t.Parallel()
+
+	accessModifiers := []string{
+		"pub",
+		"access(account)",
+		"access(contract)",
+	}
+
+	declarationKinds := []string{
+		"let",
+		"var",
+	}
+
+	valueKinds := []string{
+		"struct",
+		"resource",
+	}
+
+	runTest := func(access string, declaration string, valueKind string) {
+		testName := fmt.Sprintf("%s %s %s", access, valueKind, declaration)
+
+		assignmentOp := "="
+		var destroyStatement string
+		if valueKind == "resource" {
+			assignmentOp = "<- create"
+			destroyStatement = "destroy foo"
+		}
+
+		t.Run(testName, func(t *testing.T) {
+			_, err := ParseAndCheckWithOptions(t,
```

Comment:
Consider parallelizing the subtests here and in the other tests:
```suggestion

			t.Parallel()

			_, err := ParseAndCheckWithOptions(t,
```